File: r/stencila/html-node.R
#' @include extension.R
NULL

#' A HTML node
#'
#' Used primarily for retrieving elements of a stencil
#' using stencil methods `select` and `filter`. No explicit
#' constructor is provided.
#'
#' @name HtmlNode
#' @export
setRefClass(
	'HtmlNode',
    contains = 'Extension',
	methods = list(
		attr = function(name) method_(.self,'HtmlNode_attr',name),
		text = function() method_(.self,'HtmlNode_text'),
        select = function(selector) method_(.self,'HtmlNode_select',selector)
	)
)

File: r/stencila/src/html-node.cpp
#include <stencila/html.hpp>
using namespace Stencila;

#include "stencila.hpp"

STENCILA_R_FUNC HtmlNode_attr(SEXP self, SEXP name){
	STENCILA_R_BEGIN
		return wrap(from<Html::Node>(self).attr(
			as<std::string>(name)
		));
	STENCILA_R_END
}

STENCILA_R_FUNC HtmlNode_text(SEXP self){
	STENCILA_R_BEGIN
		return wrap(from<Html::Node>(self).text());
	STENCILA_R_END
}

STENCILA_R_FUNC HtmlNode_select(SEXP self,SEXP selector){
	STENCILA_R_BEGIN
		Html::Node* node = new Html::Node;
		*node = from<Html::Node>(self).select(
			as<std::string>(selector)
		);
		return to<Html::Node>(node,"HtmlNode");
	STENCILA_R_END
}
